climpred/bootstrap.py

```diff
@@ -497,16 +590,66 @@ def bootstrap_compute(
     # get comparison function
     comparison = get_comparison_class(comparison, ALL_COMPARISONS)
 
-    # Perfect Model requires `same_inits` setup.
-    isHindcast = True if comparison in HINDCAST_COMPARISONS else False
+    # Perfect Model requires `same_inits` setup
+    isHindcast = True if comparison.name in HINDCAST_COMPARISONS else False
     reference_alignment = alignment if isHindcast else 'same_inits'
 
-    for i in range(iterations):
-        # resample with replacement
-        smp_hind = _resample(hind, resample_dim)
-        # compute init skill
-        init_skill = compute(
-            smp_hind,
+    if hist is None:  # PM path, use verif = control
+        hist = verif
+
+    # slower path for hindcast and resample_dim init
+    if resample_dim == 'init' and isHindcast:
+        warnings.warn(f'resample_dim=`init` will be slower than resample_dim=`member`.')
+        (
+            bootstrapped_init_skill,
+            bootstrapped_uninit_skill,
+            bootstrapped_pers_skill,
+            pers_output,
+        ) = _bootstrap_hindcast_over_init_dim(
+            hind,
+            hist,
+            verif,
+            dim,
+            resample_dim,
+            iterations,
+            metric,
+            comparison,
+            alignment,
+            compute,
+            reference_compute,
+            resample_uninit,
+            reference_alignment,
+            **metric_kwargs,
+        )
+    else:  # faster resampling skill: first _resample_iterations_idx, then compute skill
+        bootstrapped_hind = _resample_iterations_idx(hind, iterations, resample_dim)
+        if not isHindcast:
+            # create more members than needed in PM to make the uninitialized
+            # distribution more robust
+            members_to_sample_from = 50
+            repeat = members_to_sample_from // hind.member.size + 1
+            uninit_hind = xr.concat(
+                [resample_uninit(hind, hist) for i in range(repeat)], dim='member'
+            )
+            uninit_hind['member'] = np.arange(1, 1 + uninit_hind.member.size)
+            # fix dask _resample_iterations_idx issue: load and then rechunk
+            if dask.is_dask_collection(uninit_hind):
+                uninit_hind = uninit_hind.load().chunk({'lead': 'auto'})
```

Comment:
The new fix

Reply:
I don't think circumventing the problem is a good idea
If we print 
```
        print(idx)
        print(x.shape)
        print(idx.shape)
```

With your circumvention
```
# idx
# [[[[2 0]]]
#  [[[1 0]]]
#  [[[1 2]]]]

#shapes
# (3, 20, 12, 1)
# (3, 1, 1, 2)
```

Without your circumvention, we can see that idx may not be correct?
```
# idx
# [[[[33  3]]]
#  [[[50 40]]]
#  [[[44 31]]]]

# shapes
# (3, 20, 12, 1)
# (3, 1, 1, 2)
```

Reply:
there seems to be a mixup between the 3 member call vs 51 member call
```
member
[[0 0 0]
 [1 2 1]]
<xarray.DataArray (iteration: 2, member: 3)>
array([[0, 0, 0],
       [1, 2, 1]])
Coordinates:
  * iteration  (iteration) int64 0 1
  * member     (member) int64 1 2 3
idx
[]
member
[[46 42 23 13 15 17 25 30 36  9 22 37 35 47 26 16  0  6 24 32 14 48  8 19
   1  7 11 40 27  4 21 34 44 18 29 33 49 41 38 39  3  2 20 50 45  5 28 31
  12 43 10]
 [17 49  3 36 41 21 22 25  4 33 14 39 19  9 12 35 38  1  6 47 45 26 37 13
  32 20 11 24  5 23 31  7 40 30 16 15 29  0 42 48 18 46  8 44 50 43 34 28
   2 27 10]]
<xarray.DataArray (iteration: 2, member: 51)>
array([[46, 42, 23, 13, 15, 17, 25, 30, 36,  9, 22, 37, 35, 47, 26, 16,
         0,  6, 24, 32, 14, 48,  8, 19,  1,  7, 11, 40, 27,  4, 21, 34,
        44, 18, 29, 33, 49, 41, 38, 39,  3,  2, 20, 50, 45,  5, 28, 31,
        12, 43, 10],
       [17, 49,  3, 36, 41, 21, 22, 25,  4, 33, 14, 39, 19,  9, 12, 35,
        38,  1,  6, 47, 45, 26, 37, 13, 32, 20, 11, 24,  5, 23, 31,  7,
        40, 30, 16, 15, 29,  0, 42, 48, 18, 46,  8, 44, 50, 43, 34, 28,
         2, 27, 10]])
Coordinates:
  * iteration  (iteration) int64 0 1
  * member     (member) int64 1 2 3 4 5 6 7 8 9 ... 43 44 45 46 47 48 49 50 51
idx
[]
```

Reply:
So how to fix this copy issue?